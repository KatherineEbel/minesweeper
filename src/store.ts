import { configureStore } from '@reduxjs/toolkit'
import {reducer as minesweeper} from 'modules/MinesweeperRedux'

export const store = configureStore({
  reducer: {
    minesweeper
  }
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch