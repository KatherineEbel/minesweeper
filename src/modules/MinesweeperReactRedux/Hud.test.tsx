import {render, screen} from '@testing-library/react'
import HudRedux from 'modules/MinesweeperReactRedux/Hud'
import React from 'react'
import {Provider} from 'react-redux'
import {store} from 'store'

test('Hud renders correctly', () => {
  const {asFragment} = render(
    <Provider store={store}>
      <HudRedux/>
    </Provider>
  )
  expect(screen.getByRole('combobox')).toHaveValue('beginner')
  expect(asFragment()).toMatchSnapshot()
})