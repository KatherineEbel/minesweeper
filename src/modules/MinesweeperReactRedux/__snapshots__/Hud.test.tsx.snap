// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hud renders correctly 1`] = `
<DocumentFragment>
  <header
    class="css-1xy6mwg"
  >
    <p
      class="css-1ke0xfb"
      role="timer"
    >
      000
    </p>
    <select
      class="css-1a3ea01"
    >
      <option
        class=" css-9cl0yr"
        value="beginner"
      >
        beginner
      </option>
      <option
        class=" css-9cl0yr"
        value="intermediate"
      >
        intermediate
      </option>
      <option
        class=" css-9cl0yr"
        value="expert"
      >
        expert
      </option>
    </select>
    <button
      class="css-12o4q79"
    >
      ðŸ™‚
    </button>
    <p
      class="css-1ke0xfb"
      role="counter"
    >
      010
    </p>
  </header>
</DocumentFragment>
`;
